packages/rich-text/src/register-format-type.js

```diff
@@ -52,6 +52,55 @@ export function registerFormatType( name, settings ) {
 		return;
 	}
 
+	if (
+		typeof settings.tagName !== 'string' ||
+		settings.tagName === ''
+	) {
+		window.console.error(
+			'Format tag names must be a string.'
+		);
```

Comment:
Not for this PR: Maybe we could leverage `yup` at some point for this and block registration, a validation schema seems like a good enhancement.